applications/OptimizationApplication/tests/test_optimization_info.py

```diff
@@ -0,0 +1,96 @@
+
+import KratosMultiphysics as Kratos
+
+# Import KratosUnittest
+import KratosMultiphysics.KratosUnittest as kratos_unittest
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_info import OptimizationInfo
+
+class TestOptimizationInfo(kratos_unittest.TestCase):
+    class TestRoutine(Kratos.Process):
+        def __init__(self):
+            super().__init__()
+
+    def test_AdvanceSolutionStepGet(self):
+        optimization_info = OptimizationInfo()
+        optimization_info.SetBufferSize(3)
+
+        optimization_info["step"] = 1
+        self.assertEqual(optimization_info["step"], 1)
+
+        optimization_info.AdvanceSolutionStep()
+        optimization_info["step"] = 2
+        self.assertEqual(optimization_info["step"], 2)
+        self.assertEqual(optimization_info.GetSolutionStepData(1)["step"], 1)
+
+        optimization_info.AdvanceSolutionStep()
+        optimization_info["step"] = 3
+        self.assertEqual(optimization_info["step"], 3)
+        self.assertEqual(optimization_info.GetSolutionStepData(1)["step"], 2)
+        self.assertEqual(optimization_info.GetSolutionStepData(2)["step"], 1)
+
+        optimization_info.AdvanceSolutionStep()
+        optimization_info["step"] = 4
+        self.assertEqual(optimization_info["step"], 4)
+        self.assertEqual(optimization_info.GetSolutionStepData(1)["step"], 3)
+        self.assertEqual(optimization_info.GetSolutionStepData(2)["step"], 2)
+        self.assertEqual(optimization_info.GetSolutionStepData(3)["step"], 4)
```

Comment:
```suggestion
        self.assertEqual(optimization_info.GetSolutionStepData(3)["step"], 1)
```

Why do we expect a 4 here? Shouldn't it be a 1 to follow the pattern?

Reply:
This is to check cyclic nature of the buffer. It is initialized with buffer size 3 in line 15, so the indices 0, 1, 2 refers to different buffer positions. But the index 3 is same as index 0 because of the cyclic nature. Thats why we should get a "4" in this one, and not "1".

Reply:
This is converted to an error. Now if solution step is requested which is out of the buffer size, an error is thrown.